lib/wifi_user/use_case/sns_notification_handler.rb

```diff
@@ -38,9 +38,15 @@ def handle_signup_request(ses_notification)
     from_address = ses_notification['mail']['commonHeaders']['from'][0]
     logger.info("Handling signup request from #{from_address}")
 
-    ::WifiUser::UseCase::EmailSignup
-      .new(user_model: WifiUser::Repository::User.new)
-      .execute(contact: from_address)
+    ::WifiUser::UseCase::EmailSignup.new(
+      user_model: WifiUser::Repository::User.new,
+      whitelist_checker: WifiUser::UseCases::CheckIfWhitelistedEmail.new(
+        gateway: Common::Gateway::S3ObjectFetcher.new(
+          bucket: ENV.fetch('S3_SIGNUP_WHITELIST_BUCKET'),
```

Comment:
I'm not too keen on the idea of a use case deciding how a configuration should be loaded.
Ideally the whitelist_checker usecase is initialised outside of the sns-notification-handler

Reply:
actually thinking about it, I think the email signup use case should also be initialised outside this use case.
While usecases can use one-another, I don't think they should be aware of one another.

Reply:
True, app.rb used to be the file that composed and envoked all these classes.

This sns_notification_handler is not a real use case, more of a facade class and I wonder if we need it at all.

Totally agree, once again refactor needed, would suggest separate PR

Reply:
Agree with @cybojenix on how composition "should" work - it's ideal for another class to have responsibility for composing things correctly.

I'm still getting my thoughts together on how configuration - especially through environment variables - interacts with the CA constructs. I've used them raw elsewhere, putting behind a gateway feels excessive, passing in from `app.rb` might be correct.

Agree with @emileswarts that any of this work could be done in a follow-up PR.

Reply:
Further thoughts: `handler` as a name could be a smell. The responsibility of this class takes a few reads to understand, and feels a little muddled.

This PR isn't changing that, though - I think the change that _is_ being made is a good one.